geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloClientConfig.scala

```diff
@@ -20,14 +26,19 @@ import scala.util.Try
 /**
  * Client configuration options, loaded from a config file
  *
- * @param instance instance name
- * @param zookeepers zookeeper connect string
- * @param zkTimeout zookeeper timeout
+ * @param config Accumulo client configuration
+ * @param hasInstance instance id or name set in config file
+ * @param instance instance name (not actually used during configuration)
+ * @param zookeepers zookeeper connect string (not actually used during configuration)
+ * @param zkTimeout zookeeper timeout (not actually used during configuration)
  * @param authType  authentication type ("password", "kerberos" or authentication token class)
  * @param principal authentication principal (username, keytab file, etc)
  * @param token authentication token (password, etc)
  */
+@SuppressWarnings(Array("deprecated"))
 case class AccumuloClientConfig(
+    config: ClientConfiguration,
+    hasInstance: Boolean,
```

Comment:
technically changing this class signature will break semantic versioning in a bug fix. It's kind of an internal class though... @jnh5y do you have any thoughts on this? if we change the target branch to main then there's no problem.

Reply:
Would package-private be sufficient?

Reply:
OK, I restored the original default constructor, made the new fields private, and made the accessors package-private so public API should remain unchanged.

Reply:
ah, great thanks!